src/harness/testcases/WINNF_FT_S_MES_testcase.py

```diff
@@ -0,0 +1,150 @@
+    device_i = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_i.json')))
+    devices = [device_a, device_c, device_e, device_f, device_g, device_i]
+
+    for device in devices:
+        device['measCapability'] = ['RECEIVED_POWER_WITH_GRANT']
+        # Inject FCC and user IDs
+        self._sas_admin.InjectFccId({'fccId': device['fccId']})
+        self._sas_admin.InjectUserId({'userId': device['userId']})
+
+    device_f['measCapability'] = ['RECEIVED_POWER_WITHOUT_GRANT']
+    cbsd_ids = []
+
+    # Register devices
+    request = {'registrationRequest': devices}
+    response = self._sas.Registration(request)['registrationResponse']
+    # Check registration response
+    self.assertEqual(len(response), 6)
+    for resp in response:
+        self.assertTrue('cbsdId' in resp)
+        self.assertEqual(resp['response']['responseCode'], 0)
+        self.assertFalse('RECEIVED_POWER_WITH_GRANT' in resp['measReportConfig'])
+        cbsd_ids.append(resp['cbsdId'])
+    del request, response
+    # Request grant
+    grant_request = []
+    for cbsd_id in cbsd_ids:
+        grant = json.load(
+            open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+        grant['cbsdId'] = cbsd_id
+        grant_request.append(grant)
+    request = {'grantRequest': grant_request}
+    response = self._sas.Grant(request)['grantResponse']
+
+    grant_ids = []
+    # Check grant response
+    for response_num, resp in enumerate(response):
+        self.assertEqual(resp['cbsdId'], cbsd_ids[response_num])
+        self.assertTrue('grantId' in resp)
+        self.assertEqual(resp['response']['responseCode'], 0)
+        grant_ids.append(resp['grantId'])
+    del request, response
+    
+    # Trigger to request measurement report for all subsequent heartbeat request
+    self._sas_admin.TriggerMeasurementReportHeartbeat({'measReportConfig':[]})
+    
+    # First heartbeat without measReport
+    heartbeat_request = []
+    for cbsd_id, grant_id in zip(cbsd_ids, grant_ids):
+        heartbeat_request.append({
+            'cbsdId': cbsd_id,
+            'grantId': grant_id,
+            'operationState': 'GRANTED'
+        })
+    request = {'heartbeatRequest': heartbeat_request}
+    response = self._sas.Heartbeat(request)['heartbeatResponse']
+    # Check first heartbeat response
+    for response_num, resp in enumerate(response):
+        self.assertEqual(resp['response']['responseCode'], 0)
+        self.assertEqual(resp['cbsdId'], cbsd_ids[response_num])
+        self.assertEqual(resp['grantId'], grant_ids[response_num])
+        transmit_expire_time = datetime.strptime(resp['transmitExpireTime'],
+                                             '%Y-%m-%dT%H:%M:%SZ')
+        self.assertLess(datetime.utcnow(), transmit_expire_time)
+        if response_num != len(response) - 1:
+            self.assertEqual(resp['measReportConfig'], 'RECEIVED_POWER_WITH_GRANT')
```

Comment:
(Just like Line 71)
self.assertTrue('RECEIVED_POWER_WITH_GRANT' in resp['measReportConfig'])